### DISCUSSION TITLE
Github Page Not Showing Up on Google Search

### DISCUSSION THREAD

**Discussion by jaehyunhan** [community/community#45850] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi! I created and published a GitHub page last month, but it is still not showing up when I Google it. When I google my name + "Github", it gives me my repository, not the actual site. Could someone please help and/or let me know how to fix this? Thanks!

**Comment by jge162** [comment]
What are the tags that you are using? They aide in the discovery of the repo.

**Reply by jge162** [reply]
I looked at your GitHub. I don’t see any tags.

**Comment by MostafaWaleed0** [comment]
1. Verify your Github page in Google Search Console: This will help Google know that your site exists and is ready to be indexed.
2. Improve your site content: Make sure your Github page has enough content, is well-written, and uses relevant keywords.
3. Get other sites to link to your Github page: This will improve your page's ranking and make it easier for Google to find and index it.
4. Wait: Google may take several weeks to index new pages, so be patient and check back later.
5. Use descriptive page titles and meta descriptions: This will make it easier for Google to understand what your page is about and show it in search results.
6. Try searching with the full URL of your Github page instead of just your name + "Github".

**Comment by 2MuchC0ff33** [comment]
There could be several reasons why your GitHub Pages site needs to appear in Google search results. Here are a few steps you can take to improve the visibility of your website:

1. Check that your website has been published: Check that your site has been published and is accessible at the correct URL. Your GitHub Pages site can be found at "username.github.io/repository-name".
2. Submit your website to Google Search Console: Submitting your website to Google Search Console can assist Google in better understanding your website and increasing its visibility in search results. You can submit your site by taking the following steps: https://search.google.com/search-console
3. Improve your site's search engine optimisation: Make sure your site's title is clear and descriptive, and use relevant keywords in the content and meta tags. More information about optimising your site for search engines can be found at https://moz.com/learn/seo/meta-description.
4. Could you share your site on social media, include a link to your site in your email signature, and reach out to your network to help promote and improve your site's visibility steps should improve your site's visibility in Google search results. Still, it may take some time for your site to appear.

Just to let you know, Nagle's algorithm for ranking pages is complex, and there is no guaranteed way to appear quickly in search results. On the other hand, following best practices for SEO and promoting your site can help improve your chances of appearing in search results.

**Comment by nschorgh** [comment]
That seems to be a fatal shortcoming of GitHub. I have the same problem and all the workarounds lead to only marginal improvements.
